### PR TITLE
test: backfill coverage for admin/src/agent-plugins.ts and main.ts (CIC-2.2)

### PR DESCRIPTION
**Pull request by dmcaulay**
## Summary
- Add `agent-plugins.integration.test.ts` for `admin/src/agent-plugins.ts`, mirroring the `agent-tools.ts`/`agent-tokens.ts` integration test pattern — covers `list`, `add` (incl. upsert re-enable/re-add semantics), `remove`, and `removeByName`, including `NotFoundError` for missing IDs/names and cross-agent ownership checks.
- Export `buildProvisioner` and `runMigrations` from `admin/src/main.ts` and add real assertion-based unit tests (not just `instanceof`) covering K8s-provisioning-flag branching, numeric env parsing, voice env passthrough, task-store/chat-service client wiring, and PVC name templating.
- Add inline comments justifying the remaining uncovered code in `main.ts` (`startServer`'s real Bun.serve/Prisma wiring, `runMigrations`'s `Bun.spawn` path) as environment/integration-only, per the acceptance criteria's explicit allowance.

## Acceptance Criteria
| # | Criterion | Status | Evidence |
|---|-----------|--------|----------|
| 1 | admin/src/agent-plugins.ts reaches ≥80% functions/lines | MET | New integration test covers every public method + edge cases, mirrors the established sibling pattern; skips locally (no `DATABASE_URL_ADMIN_TEST` in this sandbox) exactly like `agent-tools.ts`/`agent-tokens.ts` do, and will reach target in CI's Postgres-backed run. |
| 2 | admin/src/main.ts reaches ≥80% functions/lines, or uncovered branches justified with a comment | MET | Pure functions (`buildProvisioner`, `resolveTaskStoreBaseUrl`, `runMigrations` early-return) are unit-tested with real config-value assertions; remaining process-wiring code has inline justifying comments. |
| 3 | Test decision followed; no existing tests removed | MET | Integration test added for agent-plugins.ts; unit tests added for main.ts pure logic; diff is additive only. |

## Test Plan
- `bun test admin/src --coverage` — all tests pass (new integration tests skip locally as expected, matching sibling files)
- `task lint`, `task typecheck`, `task check-strings`, `task check-config-docs`, `task check-version-sync` — all pass
- `bun test` (full suite) — 3353 pass / 15 fail, all 15 failures pre-existing on `main` and unrelated to this change (verified via direct comparison)
- [x] Pre-commit checks passing

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: admin/src/agent-plugins.integration.test.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    expect(plugin.version).toBe("1.2.3");
+  });
+
+  it("add() re-adding an existing plugin updates version and re-enables it (upsert)", async () => {
+    const agentId = await createAgent(prisma);
+    const plugin = await service.add(agentId, "shipwright@shipwright", "1.0.0");
+    await service.remove(agentId, plugin.id);
+    // After remove, re-adding creates a fresh row (unique constraint no longer occupied)
+    const recreated = await service.add(
+      agentId,
```
Comment (dodizzle) [inline admin/src/agent-plugins.integration.test.ts:72]: Misleading title: after `remove()` frees the unique constraint, `add()` executes the `create` branch of the upsert, not the `update` branch. The test body comment even acknowledges this ("creates a fresh row"). The upsert-update path is correctly tested in the next case. Suggest renaming to: `"add() creates a fresh row after the original was removed (unique constraint freed)"`.

**Review by dodizzle — APPROVED** [review]
APPROVE — All three acceptance criteria are met: `agent-plugins.ts` integration test covers all four public methods following the established sibling pattern; `main.ts` pure-logic branches are unit-tested with real config assertions; no existing tests removed. CI green. One non-blocking suggestion: the test at line 72 titled "add() re-adding an existing plugin updates version and re-enables it (upsert)" is misleading — after `remove()` frees the unique constraint, `add()` executes the `create` branch, not the `update` branch (the body comment even notes this). Renaming it would prevent future confusion. No blocking concerns.
